### PR TITLE
Decompile DRA and RIC EntityHolyWaterFlame

### PR DESCRIPTION
**Pull request by bismurphy**
The arrays on the stack are weird. Couldn't have solved them without @ser-pounce , thanks again for yet another function of collaboration!

### CODE REVIEW

File: src/dra/843B0.c [hunk 1]
```diff
@@ -414,8 +414,126 @@ void EntityHolyWater(Entity* entity) {
...
+    s16* primYPtr;
 
+    primUBase = D_800B0688[(g_GameTimer & 7)].x;
+    primVBase = D_800B0688[(g_GameTimer & 7)].y;
+    switch (self->step) {
+    case 0:
+        randR = (rand() & 0x1F) | 0x40;
+        randG = (rand() & 0x1F) | 0x80;
+        randB = (rand() & 0x1F) | 0x60;
+        self->primIndex = AllocPrimitives(4U, 4);
```
Comment (Xeeynamo) [inline src/dra/843B0.c:443]:
```suggestion
        self->primIndex = AllocPrimitives(PRIM_GT4, 4);
```

File: src/ric/2C4C4.c [hunk 2]
```diff
@@ -154,7 +154,156 @@ void EntityHolyWater(Entity* self) {
...
+    s16 upperParams = self->params >> 8;
+
+    primUBase = D_80155D9C[(g_GameTimer & 7)].x;
+    primVBase = D_80155D9C[(g_GameTimer & 7)].y;
+    switch (self->step) {
+    case 0:
+        randR = (rand() & 0x3F) + 0x2F;
+        randG = (rand() & 0x3F) + 0x6F;
+        randB = (rand() & 0x7F) + 0x6F;
+        self->primIndex = g_api.AllocPrimitives(4U, 4);
```
Comment (Xeeynamo) [inline src/ric/2C4C4.c:186]:
```suggestion
        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, 4);
```

**Review by Xeeynamo — APPROVED** [review]
Just a very minor thing on the `AllocPrimitives`. But everything else looks fantastic! Congrats to you both :clap:

File: src/dra/843B0.c (hunk 1 above)
Comment (bismurphy) [reply src/dra/843B0.c:443]: Ah, of course. Good catch! Fixing now. Thanks!

**Pull request merged**